### PR TITLE
Update glide.yaml

### PR DESCRIPTION
**Pull request by richardartoul**

### CODE REVIEW

File: linters/badtime/glide.yaml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 package: github.com/m3db/build-tools/timemaplint
 import:
 - package: github.com/kisielk/gotool
-  version: d6ce6262d87e3a4e153e86023ff56ae771554a41
-- package: github.com/stretchr/testify
-  version: v1.1.4
+  version: master
```
Comment (prateek) [inline linters/badtime/glide.yaml]: why not pin to a specific revision?

**Comment by richardartoul** [comment]
@prateek akshay asked me to not spin specific SHA's if we're gonna release this to the rest of the company

**Comment by prateek** [comment]
It's not possible to get reproducible builds without pinning to a particular version of the dependencies. If people want the ability to stay on the bleeding edge, I'm fine with leaving this repo's master un-pinned. But if we do that, we should follow a tagging of releases (preferably using semver) where we do pin to a SHA1.

**Comment by richardartoul** [comment]
@prateek Once this is landed I was planning on releasing a tagged version 1.0 which will still have the glide.lock pinned so we will have reproducible builds, right?

**Comment by prateek** [comment]
Nah, a `glide update -v` will update stuff that isn't pinned in the `glide.yaml` if there's a new version available.

**Comment by richardartoul** [comment]
That seems like a different definition of reproducible build. You're saying it won't be reproducible once you start updating dependencies which doesn't seem that bad

**Comment by prateek** [comment]
The same thing applies when do I a `glide install -v`. So for e.g. in travis-ci runs, it's always going to be pulling the latest version.

**Comment by prateek** [comment]
@richardartoul tried the experiment offline, `glide install -v` respects the version in the lock file for a clean install too. Considering this is a binary we can leave `master` in glide.yaml then.

**Pull request merged**